### PR TITLE
Add login, logout and edit-user details

### PR DESCRIPTION
**Pull request by amanCoder110599**
This PR is to:
- Add feature for login and logout.
- Add feature to redirect the user to edit profile page if the profile is incomplete.

The feature is deployed (https://summer20-sps-95.el.r.appspot.com/).

### CODE REVIEW

File: project/src/main/java/com/google/sps/data/User.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  private final String phone;
+  private final String rating;
+
+  public User(String name, String email, String phone, String rating){
+    this.name = name;
+    this.email = email;
+    this.phone = phone;
+    this.rating = rating;
+  }
+}
```
Comment (palak2001) [inline project/src/main/java/com/google/sps/data/User.java:15]: add newline

File: project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      //If the current user already exist but has dummy details
+      else{
+        status.put("editDetails", true);
+      }
+    }
+
+    else {
+      status.put("loggedIn", false);
+      String urlToRedirectToAfterUserLogsIn = "/index.html";
```
Comment (palak2001) [inline project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:73]: Unsymmetrical indentation of else

File: project/src/main/webapp/editProfile.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    document.getElementById("email").value = email;
+    document.getElementById("phone").value = phone;
+    document.getElementById("rating").value = rating;
+    
+  });
+}
+
+window.onload = () => {
+  loadEditProfilePage();
+}
```
Comment (TheArijitSaha) [inline project/src/main/webapp/editProfile.js:23]: nit: newline before EOF

File: project/src/main/webapp/script.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  loginDiv.appendChild(loginAsk);
+  loginDiv.appendChild(loginButton);
+
+  document.querySelector('div.' + 'LoginLogoutDiv')
+          .appendChild(loginDiv);
+}
+
+window.onload = () => {
+  loadLoginLogoutButton();
+}
```
Comment (TheArijitSaha) [inline project/src/main/webapp/script.js:44]: nit: newline before EOF

File: project/pom.xml [hunk 5]
```diff
@@ -43,6 +43,18 @@
...
     </dependency>
 
+    <dependency>  
+      <groupId>com.googlecode.json-simple</groupId>  
+      <artifactId>json-simple</artifactId>  
+      <version>1.1</version>  
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-datastore</artifactId>
```
Comment (TheArijitSaha) [inline project/pom.xml:53]: Why are we adding this dependency separately when Appengine dependency already provides interface to datastore.

File: project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;    
+
+@WebServlet("/login-status")
+public class LoginStatusServlet extends HttpServlet {
+
+  /**
+   * @return JSON response with the status of the current logged-in user and if the user details are complete.
+   */
+
```
Comment (TheArijitSaha) [inline project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:24]: nit: you may remove this empty line. Javadoc and function for which the doc is meant should not be separated by vertical whitespace.

File: project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      status.put("loggedIn", true);
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/index.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      status.put("logoutUrl", logoutUrl);
+      status.put("email", userEmail);
+
+      Datastore datastore = DatastoreOptions.newBuilder().setProjectId("summer20-sps-95").build().getService();
```
Comment (TheArijitSaha) [inline project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:38]: This becomes a lot smoother and cleaner if you use Datastore API provided by Appengine like it said in sps tutorials. The binding to project ID is done automatically and no need of hardcoding project Id in code is needed.

Any specific reason as to why we are using datastore this way?

File: project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+@WebServlet("/login-status")
+public class LoginStatusServlet extends HttpServlet {
+
+  /**
+   * @return JSON response with the status of the current logged-in user and if the user details are complete.
+   */
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JSONObject status = new JSONObject();
```
Comment (TheArijitSaha) [inline project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:27]: As we are already using Gson, you may use Gson's JsonObject and remove dependency on json-simple altogether. Gson's JsonObject has pretty similar structure:
```
Gson gson = new Gson();
JsonObject json = new JsonObject();
json.addProperty("logoutUrl", logoutUrl);                         // To add a simple json property
json.add("someObject", gson.toJsonTree(someObject));              // To add an object as a json property
```

File: project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java (hunk 7 above)
Comment (amanCoder110599) [reply project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:38]: I was having some issues using the datastore service in the query part. On the web, I could mostly find this way of defining datastore and its query is fairly easier and a bit efficient in my opinion as it does not require the overhead for Filter.
Later on, if I found the solution to the query, I will replace this in subsequent commits.

Comment (TheArijitSaha) [reply project/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:39]: Great!

**Pull request merged**